tools/azure-sdk-tools/devtools_testutils/azure_recorded_testcase.py

```diff
@@ -261,3 +189,109 @@ def generate_sas(self, *args, **kwargs):
+            _LOGGER.info(
+                "Environment variable AZURE_TEST_USE_AZD_AUTH set to 'true'. Using AzureDeveloperCliCredential."
+            )
+            from azure.identity import AzureDeveloperCliCredential
+
+            if is_async:
+                from azure.identity.aio import AzureDeveloperCliCredential
+            return AzureDeveloperCliCredential(**kwargs)
+
+        # Service principal authentication
+        if tenant_id and client_id and secret:
+            _LOGGER.info(
+                "Service principal client ID, secret, and tenant ID detected. Using ClientSecretCredential.\n"
+                "For user-based auth, set AZURE_TEST_USE_PWSH_AUTH or AZURE_TEST_USE_CLI_AUTH to 'true'."
+            )
+            from azure.identity import ClientSecretCredential
+
+            if is_async:
+                from azure.identity.aio import ClientSecretCredential
+            return ClientSecretCredential(tenant_id=tenant_id, client_id=client_id, client_secret=secret, **kwargs)
+
+        # If AzurePipelinesCredential is detected, use it.
+        service_connection_id = os.environ.get("AZURESUBSCRIPTION_SERVICE_CONNECTION_ID")
+        client_id = os.environ.get("AZURESUBSCRIPTION_CLIENT_ID")
+        tenant_id = os.environ.get("AZURESUBSCRIPTION_TENANT_ID")
+        system_access_token = os.environ.get("SYSTEM_ACCESSTOKEN")
+        if service_connection_id and client_id and tenant_id and system_access_token:
+            from azure.identity import AzurePipelinesCredential
+            if is_async:
+                from azure.identity.aio import AzurePipelinesCredential
+            return AzurePipelinesCredential(
+                tenant_id=tenant_id,
+                client_id=client_id,
+                service_connection_id=service_connection_id,
+                system_access_token=system_access_token,
+                **kwargs
+            )
+        # This is for testing purposes only, to ensure that the AzurePipelinesCredential is used when available
+        # else:
+        #     raise ValueError(
+        #         "Environment variables not set for service principal authentication. "
+        #         f"service_connection_id: {service_connection_id}, client_id: {client_id}, tenant_id: {tenant_id}, system_access_token: {system_access_token}"
+        #     )
+        # Fall back to DefaultAzureCredential
+        from azure.identity import DefaultAzureCredential
+        if is_async:
+            from azure.identity.aio import DefaultAzureCredential
+        return DefaultAzureCredential(exclude_managed_identity_credential=True, **kwargs)
+
+    # For playback tests, return credentials that will accept playback `get_token` calls
+    else:
+        if is_async:
+            if is_live:
+                raise ValueError(
+                    "Async live doesn't support mgmt_setting_real, please set AZURE_TENANT_ID, "
+                    "AZURE_CLIENT_ID, AZURE_CLIENT_SECRET"
+                )
```

Comment:
Since the `else` is already for the `is_live` check, we can remove this check. We can also then remove the `is_live = is_live()` assignment on line 204 since leaves us with only one use of the check.
```suggestion
```